### PR TITLE
Cross Entropy Loss

### PR DESCRIPTION
**Pull request by AntoineDubois**
This pull request introduces the Cross Entropy Loss function to mlpack, expanding it beyond the current binary categorical entropy. The Cross Entropy Loss function is widely used in deep reinforcement learning to train policy networks. This addition is a step toward implementing AlphaZero.

# Features
	1.	Implementation of the cross entropy loss function
	2.	Unit testing of the cross entropy loss function
	3.	Removal of the alias `CrossEntropyErrorType` and `CrossEntropyError ` of the binary cross entropy class `BCELossType`, which previously suggested that binary cross-entropy could support not categorical and multidimensional inputs.

### CODE REVIEW

File: src/mlpack/methods/ann/loss_functions/binary_cross_entropy_loss.hpp [hunk 1]
```diff
@@ -91,17 +91,10 @@ class BCELossType
...
 typedef BCELossType<arma::mat> BCELoss;
 
-/**
- * Alias of BCELossType.
- */
-typedef BCELossType<arma::mat> CrossEntropyError;
-
-template<typename MatType = arma::mat>
-using CrossEntropyErrorType = BCELossType<MatType>;
-
```
Comment (shrit) [inline src/mlpack/methods/ann/loss_functions/binary_cross_entropy_loss.hpp:101]: Is there any reason this is being removed ?
